### ISSUE TITLE
(One#26753137) [Navigation on actions]: Ensures buttons have discernible text.

### ISSUE THREAD

**Issue by ek68794998** [Azure/LogicAppsUX#4293]
### **User Experience:** 

User would find difficulty to identify the controls if name is not defined for controls as they won't get the context of that control and also, they won't be able to perform functionality relevant to that control.

### **Repro Steps:** 

1. Open URL "https://make.powerautomate.com/" and sign in with Valid credentials. 
2. Click on My Flows on left nav. 
3. For a flow, click to 'Edit' button and select it. 
4. Edit page will appear, navigate to 'New Step' button and select it. 
5. Choose an Action section will appear on the screen, Navigate to all Actions present in section. 
6. Run the accessibility insights for web tool. 
7. Verify whether 'Close' button has discernible text or not. 

### **Actual Result:** 

Button does not have discernible text for 'close' in Add an action window.

Observation: Screen reader is announcing as only "button" when focus moved to the control

### **Expected Result:** 

Button should have discernible text for 'Close' in Add an action window.

### **Issue:** 

Ensures buttons have discernible text (button-name - [https://accessibilityinsights.io/info-examples/web/button-name](<https://accessibilityinsights.io/info-examples/web/button-name>))

### **Target application:** 

Microsoft Power Automate - [https://make.preview.powerautomate.com/environments/839eace6-59ab-4243-97ec-a5b8fcc104e4/flows/3b6866d6-01d0-4505-a5c3-7d51330eaf7b?v3=true](<https://make.preview.powerautomate.com/environments/839eace6-59ab-4243-97ec-a5b8fcc104e4/flows/3b6866d6-01d0-4505-a5c3-7d51330eaf7b?v3=true>)

### **Element path:** 

.\_\_\_1p24v7a

### **Snippet:** 

```
<button type="button" class="fui-Button r1alrhcs ___1p24v7a fhovq9v f1p3nwhy f11589ue f1q5o8ev f1pdflbu fkfq4zb f1t94bn6 f1s2uweq fr80ssc f1ukrpxl fecsdlb fnwyq0v ft1hn21 fuxngvv fy5bs14 fsv2rcd f1h0usnq fs4ktlq f16h9ulv fx2bmrt f1omzyqd f1dfjoow f1j98vj9 fj8yq94 f4xjyn1 f1et0tmh f9ddjv3 f1wi8ngl f1sbtcvk fwiuce9 fdghr9 f15vdbe4 fwbmr0d f44c6la">
```

### **How to fix:** 

**Fix any of the following:**

- Element does not have inner text that is visible to screen readers 
- aria-label attribute does not exist or is empty 
- aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty 
- Element has no title attribute 
- Element's default semantics were not overridden with role="none" or role="presentation" 

Environment: Microsoft Edge version 121.0.0.0

\====

This accessibility issue was found using Accessibility Insights for Web 2.41.0 (axe-core 4.7.2), a tool that helps find and fix accessibility issues. Get more information & download this tool at [http://aka.ms/AccessibilityInsights](<http://aka.ms/AccessibilityInsights>).

### **MAS Reference:** 

[MAS 4.1.2 -- Name, Role, Value](<https://aka.ms/MAS4.1.2>)

### **ARIA Authoring Ref Link/Fluent Patterns Guidance /HITS Guidance:** 

[https://www.w3.org/WAI/ARIA/apg/patterns/tabs/](<https://www.w3.org/WAI/ARIA/apg/patterns/tabs/>)

### **UIA/DOM details available:** 

Yes

### **Reference Links:** 

- [Accessibility Insights!](<https://nam06.safelinks.protection.outlook.com/?url=https://accessibilityinsights.io/&data=02%7c01%7cv-poyad%40microsoft.com%7cc11478591b5a474f6d9d08d7199726c0%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c637006009300301864&sdata=ZTWYamT1yE7yko5QU35ZJLf/5ZydtDeGAKrQGUiugJs%3D&reserved=0>) - Identify accessibility bugs before check-in and make bug fixing faster and easier. 

### Are you using new designer or old designer 

New Designer (With action panels)

### Screenshots or Videos 

![MAS4 1 2 Ensures buttons have discernible text](<https://github.com/Azure/LogicAppsUX/assets/1350074/347f9e35-83bb-426d-98ab-fc06f7d97d89>)

### Browser 

- **OS Version:** Windows 11 Version 23H2 (OS Build 22631.3007) 
- **Browser:** Edge Version 120.0.2210.144 (Official build) (64-bit) 
- **\#URL:**[https://make.powerautomate.com/](<https://make.powerautomate.com/>) 
- **Screen Reader: **Narrator 

### Additional context 

[One#26753137](<https://dev.azure.com/msazure/One/_workitems/edit/26753137>)

**Comment by hartra344** [comment]
This is already done

![CleanShot 2024-03-01 at 13 34 13@2x](https://github.com/Azure/LogicAppsUX/assets/13208452/8dec6170-51b2-4e8a-aaef-44675ef06a27)
